core/shared/src/main/scala/org/http4s/multipart/Multiparts.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s.multipart
+
+import cats.effect.Sync
+import cats.syntax.all._
+
+import scala.util.Random
+
+/** An algebra for creating multipart values and boundaries.
+  *
+  * A single instance may be shared by the entire application.
+  */
+trait Multiparts[F[_]] {
+
+  /** Generate a random multipart boundary */
+  def boundary: F[Boundary]
+
+  /** Generate a multipart value from Parts with a random multipart boundary */
+  def multipart(parts: Vector[Part[F]]): F[Multipart[F]]
+}
+
+object Multiparts {
+
+  /** Creates a `scala.util.Random` and provides Multiparts around it.
+    * This instance can be shared, or is cheap enough to create closer
+    * to where Multipart values are generated.
+    */
+  def forSync[F[_]](implicit F: Sync[F]): F[Multiparts[F]] =
+    F.delay(new Random()).map(fromScalaRandom[F])
+
+  def fromScalaRandom[F[_]](random: Random)(implicit F: Sync[F]): Multiparts[F] =
```

Comment:
I think I forgot this on 0.23.x branch, we should switch to use `Random` typeclass here instead.

Reply:
Will you port this to the 0.23 branch? Or do you say that we should do it only for the main?

Reply:
We should definitely fix it on the 0.23.x branch. I can do it a bit later 👍